pkg/ext-proc/scheduling/scheduler.go

```diff
@@ -42,13 +44,40 @@ var (
 		},
 	}
 
+	// lowLatencyFilterNoLoRA is the same as lowLatencyFilter but without the LoRA cost filter.
+	lowLatencyFilterNoLoRA = &filter{
+		name:   "least queuing",
+		filter: leastQueuingFilterFunc,
+		nextOnSuccessOrFailure: &filter{
+			name:   "least KV cache percent",
+			filter: leastKVCacheFilterFunc,
+		},
+	}
+
+	// lowLatencyFilterModified defaults to lowLatencyFilterLoRA above a certain queueing threshold. LoRA affinity takes precedence below that queueing threshold.
+	lowLatencyFilterModified = &filter{
+		name:   "low queueing filter",
+		filter: lowQueuingFilterFunc,
+		nextOnSuccess: &filter{
+			name:          "affinity LoRA",
+			filter:        toFilterFunc(loRAAffinityPredicate),
+			nextOnSuccess: lowLatencyFilterNoLoRA,
+			nextOnFailure: &filter{
+				name:                   "min cost LoRA",
+				filter:                 toFilterFunc(minLoRAPredicate),
+				nextOnSuccessOrFailure: lowLatencyFilterNoLoRA,
```

Comment:
lowLatencyFilterLoRA is needed when we first prioritze lora affinity and followed by lowLatencyFilterNoLoRA (queueing  + least KV Cache). We can probably reuse lowLatencyFilterLoRA but it would be very confusing

Reply:
I had the same thought, I was also expecting that we will  reuse `lowLatencyFilterLoRA`, and so this filter will look like:

```
lowLatencyFilterModified = &filter{
  name:   "low queueing filter",
  filter: toFilterFunc((lowQueueingPodPredicate)),
  nextOnSuccessOrFailure: lowLatencyFilterLoRA
}
```

Reply:
I see, so lowLatencyFilterLoRA prioritized queuing over LoRA Affinity (this is how we had it originally) i.e. Least Queueing -> Min Cost LoRA -> Least KV Cache

Reply:
oh, I just noticed that we flip the order here compared to `lowLatencyFilterLoRA`

Here we do: LoRA -> queue length -> kv-cache
There we do: queue length -> LoRA -> kv-cache

is this by design?

Reply:
yes, so i realized the names are very confusing. So I renamed the filters. Yes its by design to make LoRA Affinity stronger.